modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/AttributedData.scala

```diff
@@ -12,5 +12,15 @@
  */
 package com.snowplowanalytics.snowplow.enrich.common.fs2
 
-/** Data intended to be sent to a sink, plus key-value pairs to be added to the message as attributes */
-final case class AttributedData[A](data: A, attributes: Map[String, String])
+/**
+ * Represents a payload, partition key and attributes of the payload
+ * @param data payload to be sent to the sink
+ * @param partitionKey field name to be used as partition key, supported by Kinesis and Kafka
+ * @param attributes key-value pairs to be added to the message as attributes, supported by PubSub and Kafka
+ * @tparam A type of the payload
+ */
+final case class AttributedData[A](
+  data: A,
+  partitionKey: String,
```

Comment:
This is a nice change, to make the partition key a field of attributed data.

However, in the master branch we currently have [this code](https://github.com/snowplow/enrich/blob/3.2.3/modules/kinesis/src/main/scala/com/snowplowanalytics/snowplow/enrich/kinesis/Sink.scala#L172-L176) for kinesis, which takes the partition key from the attributes.  Do you think we should update that kinesis code to make use of this new partitionKey field?

Reply:
I think we should update it. I intentionally left it out, not to touch enrich-kinesis in this PR. I was planning to open another PR after this one is merged. Would you prefer to address that in this PR?

Reply:
addressed in this PR